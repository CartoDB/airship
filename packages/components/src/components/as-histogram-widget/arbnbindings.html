
<!DOCTYPE html>
<html lang="en">

<head>
    <meta charset="UTF-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0">
    <meta http-equiv="X-UA-Compatible" content="ie=edge">

    <title>Airship histogram selectionChanged bug</title>

    <script src="https://cdn.jsdelivr.net/npm/lodash@4.17.11/lodash.min.js"></script>
    <script src="https://libs.cartocdn.com/carto-vl/v1.0.0/carto-vl.min.js"></script>

    <link href="https://api.tiles.mapbox.com/mapbox-gl-js/v0.50.0/mapbox-gl.css" rel="stylesheet" />
    <script src="https://api.tiles.mapbox.com/mapbox-gl-js/v0.50.0/mapbox-gl.js"></script>

    <link rel="stylesheet" href="/packages/styles/dist/airship.css">
    <script src="/packages/components/dist/airship.js"></script>
    <script src="/packages/bridge/dist/asbridge.js"></script>
</head>

<body class="as-app-body">
    <div class="as-app">
        <div class="as-content">
            <main class="as-main">
                <div class="as-map-area">
                    <div id="map"></div>
                </div>
                <div class="as-map-footer">
                  <div class="as-box">
                    <as-histogram-widget
                      show-clear
                      id="nights"
                      heading="Min Nights"
                    >
                    </as-histogram-widget>
                  </div>
                  <div class="as-box">
                    <as-histogram-widget
                      id="price"
                      heading="Price"
                      show-clear
                    >
                    </as-histogram-widget>
                  </div>
                  <div class="as-box">
                    <as-histogram-widget
                      id="lat"
                      heading="Latitude"
                      show-clear
                    >
                    </as-histogram-widget>
                  </div>
                  <div class="as-box">
                    <as-histogram-widget
                      id="lon"
                      heading="Longitude"
                      show-clear
                    >
                    </as-histogram-widget>
                  </div>
                </div>
            </main>
        </div>
    </div>

    <script>
        const map = new mapboxgl.Map({
          container: 'map',
          style: carto.basemaps.darkmatter,
          center: [-3.69, 40.38],
          zoom: 10
        });

        carto.setDefaultAuth({
          username: 'roman-carto',
          apiKey: 'default_public'
        });

        const s = carto.expressions;

        const source = new carto.source.SQL(`select * from listings`);
        const viz = new carto.Viz(`
          strokeWidth: 0.5
          width: 8
          color: red
          strokeColor: red
        `);

        const vizLayer = new carto.Layer('layer', source, viz);

        const priceWidget = document.querySelector('#price');
        const nightsWidget = document.querySelector('#nights');
        const latWidget = document.querySelector('#lat');
        const lonWidget = document.querySelector('#lon');

        const listingsLayer = new AsBridge.VLBridge(
          carto,
          map,
          vizLayer
        );
          
        listingsLayer.histogram({
          column: 'price',
          buckets: 30,
          readOnly: false,
          source,
          viz,
          widget: priceWidget,
        });
        
        listingsLayer.histogram({
          column: 'minimum_nights',
          buckets: 10,
          readOnly: false,
          source,
          viz,
          widget: nightsWidget
        });

        listingsLayer.histogram({
          column: 'latitude',
          buckets: 30,
          readOnly: false,
          source,
          viz,
          widget: latWidget,
        });

        listingsLayer.histogram({
          column: 'longitude',
          buckets: 30,
          readOnly: false,
          source,
          viz,
          widget: lonWidget,
        });

        listingsLayer.build();
          
        vizLayer.addTo(map, 'watername_ocean');
    </script>
</body>

</html>
