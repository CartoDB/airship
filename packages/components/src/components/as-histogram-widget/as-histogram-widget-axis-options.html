<!DOCTYPE html>
<html dir="ltr" lang="en">

<head>
  <base href="/">
  <meta charset="utf-8">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0">
  <title>Airship</title>
  <link rel="stylesheet" href="packages/styles/dist/airship.css">
  <script src="packages/components/dist/airship.js"></script>
  <style>
    body {
      height: 100vh;
      margin: 0;
      padding: 0;
      background: white;
    }

    as-histogram-widget {
      width: 100%;
      height: 100%;
      background: white;
    }

  </style>
</head>

<body>
    <h2 class="as-title">Y-Axis options</p>
    <div style="width: 100%; display: flex;">
      <as-histogram-widget
        id="y1"
        heading="With custom tick values"
        description="Show 20 and 25"
        show-header
        disable-interactivity>
      </as-histogram-widget>
      <as-histogram-widget
        id="y2"
        heading="With custom tick count"
        description="Show 1 tick"
        show-header
        disable-interactivity>
      </as-histogram-widget>
      <as-histogram-widget
        id="y3"
        heading="With custom tick formatter"
        description="Show ยก${value}!"
        show-header
        disable-interactivity>
      </as-histogram-widget>
    </div>
    <h2 class="as-title">X-Axis options</p>
    <div style="width: 100%; display: flex;">
      <as-histogram-widget
        id="x1"
        heading="With custom tick values"
        description="Show 20 and 25"
        show-header
        disable-interactivity>
      </as-histogram-widget>
      <as-histogram-widget
        id="x2"
        heading="With custom tick count"
        description="Show 1 tick"
        show-header
        disable-interactivity>
      </as-histogram-widget>
      <as-histogram-widget
        id="x3"
        heading="With custom tick formatter"
        description="Show ~${value}"
        show-header
        disable-interactivity>
      </as-histogram-widget>
    </div>

    <script>
      var histY1 = document.querySelector('#y1');
      var histY2 = document.querySelector('#y2');
      var histY3 = document.querySelector('#y3');
      var histX1 = document.querySelector('#x1');
      var histX2 = document.querySelector('#x2');
      var histX3 = document.querySelector('#x3');
      var all = [histY1, histY2, histY3, histX1, histX2, histX3];

      var data = [
        { start: 0, end: 10, value: 30, color: '#74d7ec' },
        { start: 10, end: 20, value: 30, color: '#ffafc7' },
        { start: 20, end: 30, value: 0 },
        { start: 30, end: 40, value: 30, color: '#ffafc7' },
        { start: 40, end: 50, value: 30, color: '#74d7ec' },
      ];

      all.forEach((widget) => widget.data = data);

      histY1.yAxisOptions = {
        values: [20, 25],
      };
      histY2.yAxisOptions = {
        ticks: 1
      };
      histY3.yAxisOptions = {
        padding: 8,
        format: (value) => `ยก${value}!`
      };
      histX1.xAxisOptions = {
        values: [20],
      };
      histX2.xAxisOptions = {
        ticks: 1
      };
      histX3.xAxisOptions = {
        format: (value) => `~${value}`
      };
    </script>
  </body>

</html>
