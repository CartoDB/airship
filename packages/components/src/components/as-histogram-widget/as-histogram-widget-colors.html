<!DOCTYPE html>
<html dir="ltr" lang="en">
  <head>
    <base href="/">
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0">
    <title>Airship</title>
    <link rel="stylesheet" href="packages/styles/dist/airship.css">
    <script type="module" src="packages/components/dist/airship/airship.esm.js"></script>
    <script nomodule="" src="packages/components/dist/airship/airship.js"></script>
  </head>

  <body>
    <div style="width: 250px;">
      <as-histogram-widget
        show-clear
        heading="Business Volume"
        description="Description">
      </as-histogram-widget>
    </div>

    <script>
      var histogramWidget = document.querySelector('as-histogram-widget');
      histogramWidget.data = [
        { start: 0, end: 10, value: 1050, color: '#FF0000' },
        { start: 10, end: 20, value: 1150, color: '#00FF00' },
        { start: 20, end: 30, value: 1250, color: 'rgb(0, 0, 255)' },
        { start: 30, end: 40, value: 1350, color: 'rgba(0, 0, 0, 0.5)'},
        { start: 40, end: 50, value: 2000 },
      ];
    </script>
  </body>
</html>
