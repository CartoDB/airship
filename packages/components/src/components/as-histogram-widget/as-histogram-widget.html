<!DOCTYPE html>
<html dir="ltr" lang="en">
  <head>
    <base href="/">
    <meta charset="utf-8">
    <meta name="viewport" content="width=device-width, initial-scale=1.0, minimum-scale=1.0, maximum-scale=5.0">
    <title>Airship</title>
    <link rel="stylesheet" href="packages/styles/dist/airship.css">
    <script type="module" src="packages/components/dist/airship/airship.esm.js"></script>
    <script nomodule="" src="packages/components/dist/airship/airship.js"></script>
    <style>
        body {
          background: #F2F2F2;
        }

        as-histogram-widget {
          width: 350px;
          margin: 16px auto;
          background: white;
        }
    </style>
  </head>

  <body>
    <as-histogram-widget
        show-clear
        heading="Business Volume"
        description="Description">
    </as-histogram-widget>

    <as-histogram-widget
      id="widget-1"
      heading="Business Volume"
      description="Description"
      error="Unexpected error"
      error-description="There is no internet connection">
    </as-histogram-widget>

    <as-histogram-widget
      id="widget-2"
      heading="Business Volume"
      description="Description"
      is-loading="true">
    </as-histogram-widget>

    <as-histogram-widget
      id="widget-3"
      heading="Business Volume"
      description="Description">
    </as-histogram-widget>

    <script>
      var histogramWidget = document.querySelector('as-histogram-widget');
      histogramWidget.data = [
        { start: 0, end: 10, value: 1050 },
        { start: 10, end: 20, value: 1150 },
        { start: 20, end: 30, value: 1250 },
        { start: 30, end: 40, value: 1350 },
        { start: 40, end: 50, value: 2000 },
      ];

      // Empty
      var noDataHistogramWidget = document.getElementById('widget-3');
      setTimeout(() => { noDataHistogramWidget.data = categories; }, 1500);
      setTimeout(() => { noDataHistogramWidget.data = []; }, 3000);
    </script>
  </body>
</html>
