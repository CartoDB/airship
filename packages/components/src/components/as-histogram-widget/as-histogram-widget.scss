@import '../../../../styles/src/core/_dev';
@import '../../../../styles/src/tooltip/tooltip';

as-histogram-widget {
  display: block;
  background: var(--histogram-widget--background-color, $color-ui-01);

  .as-histogram-widget__header {
    font: $font-subheader;
  }

  .as-histogram-widget__description {
    font: $font-body;
  }

  .as-histogram-widget__tooltip {
    @extend .as-tooltip;
    position: absolute;
    opacity: 0;

    &::after {
      opacity: 0;
    }
  }

  svg {
    display: block;

    .brush {
      .selection {
        stroke: none;
        fill: none;
      }
    }

    .bar {
      transition: all 0.3s linear;
      opacity: 0.7;
      cursor: pointer;
    }

    .yAxis .tick {
      line {
        stroke: var(--as-color-type-02, #{$color-type-02});
        opacity: 0.1;
      }

      &:first-of-type {
        line {
          opacity: 0.5;
        }
      }
    }

    text {
      fill: $color-ui-04;
    }
  }
}
