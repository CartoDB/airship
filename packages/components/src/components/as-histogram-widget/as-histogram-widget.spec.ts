import { HistogramWidget } from './as-histogram-widget';

describe('as-histogram-widget', () => {
  let histogramWidget;

  beforeEach(() => {
    histogramWidget = new HistogramWidget();
  });

  it('should build', () => {
    expect(histogramWidget).toBeTruthy();
  });
});
