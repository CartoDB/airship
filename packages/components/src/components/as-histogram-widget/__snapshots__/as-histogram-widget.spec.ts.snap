// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`as-histogram-widget Rendering should not render header when showHeader is false 1`] = `
<as-histogram-widget
  class="hydrated"
>
  
  
  <div
    class="as-histogram-widget__wrapper"
  >
    <span
      class="as-histogram-widget__tooltip"
      role="tooltip"
    >
      
    </span>
    <svg
      viewBox="0 0 248 160"
    >
      <g
        class="yAxis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="end"
        transform="translate(35, 15)"
      />
      <g
        class="xAxis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(35, 140)"
      >
        <g
          class="tick"
          opacity="1"
          transform="translate(0.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="10"
          >
            0
          </text>
        </g>
      </g>
      <g
        transform="translate(35, 15)"
      />
      <g
        class="brush"
        fill="none"
        pointer-events="all"
      >
        <rect
          class="overlay"
          cursor="crosshair"
          height="125"
          pointer-events="all"
          width="205"
          x="35"
          y="15"
        />
        <rect
          class="selection"
          cursor="move"
          fill="#777"
          fill-opacity="0.3"
          shape-rendering="crispEdges"
          stroke="#fff"
          style="display: none;"
        />
        <rect
          class="handle handle--e"
          cursor="ew-resize"
          style="display: none;"
        />
        <rect
          class="handle handle--w"
          cursor="ew-resize"
          style="display: none;"
        />
        <rect
          class="handle--custom"
          cursor="ew-resize"
          fill="#47DB99"
          height="15"
          rx="100"
          ry="100"
          style="opacity: 0;"
          width="15"
        />
        <rect
          class="handle--custom"
          cursor="ew-resize"
          fill="#47DB99"
          height="15"
          rx="100"
          ry="100"
          style="opacity: 0;"
          width="15"
        />
        <line
          class="bottomline"
          pointer-events="none"
          stroke="#47DB99"
          stroke-width="4"
          style="opacity: 0;"
          y1="140"
          y2="140"
        />
      </g>
    </svg>
  </div>
  
</as-histogram-widget>
`;

exports[`as-histogram-widget Rendering should render clear button 1`] = `
<as-histogram-widget
  class="hydrated"
>
  
  <h2
    class="as-histogram-widget__header"
  >
    
  </h2>
  <p
    class="as-histogram-widget__description as-body"
  >
    
  </p>
  <div
    class="as-histogram-widget__wrapper"
  >
    <span
      class="as-histogram-widget__tooltip"
      role="tooltip"
    >
      
    </span>
    <svg
      viewBox="0 0 248 160"
    >
      <g
        class="yAxis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="end"
        transform="translate(35, 15)"
      >
        <g
          class="tick"
          opacity="1"
          transform="translate(0,125.5)"
        >
          <line
            stroke="currentColor"
            x2="205"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-10"
          >
            0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,94.25)"
        >
          <line
            stroke="currentColor"
            x2="205"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-10"
          >
            5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,63)"
        >
          <line
            stroke="currentColor"
            x2="205"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-10"
          >
            10
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,31.75)"
        >
          <line
            stroke="currentColor"
            x2="205"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-10"
          >
            15
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,0.5)"
        >
          <line
            stroke="currentColor"
            x2="205"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-10"
          >
            20
          </text>
        </g>
      </g>
      <g
        class="xAxis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(35, 140)"
      >
        <g
          class="tick"
          opacity="1"
          transform="translate(0.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="10"
          >
            0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(51.75,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="10"
          >
            10
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(103,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="10"
          >
            20
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(154.25,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="10"
          >
            30
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(205.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="10"
          >
            40
          </text>
        </g>
      </g>
      <g
        transform="translate(35, 15)"
      >
        <rect
          class="bar"
          height="0"
          style="fill: #1785FB;"
          width="50.25"
          x="0"
          y="125"
        />
        <rect
          class="bar"
          height="0"
          style="fill: #1785FB;"
          width="50.25"
          x="51.25"
          y="125"
        />
        <rect
          class="bar"
          height="0"
          style="fill: #1785FB;"
          width="50.25"
          x="102.5"
          y="125"
        />
        <rect
          class="bar"
          height="0"
          style="fill: #1785FB;"
          width="50.25"
          x="153.75"
          y="125"
        />
      </g>
      <g
        class="brush"
        fill="none"
        pointer-events="all"
      >
        <rect
          class="overlay"
          cursor="crosshair"
          height="125"
          pointer-events="all"
          width="205"
          x="35"
          y="15"
        />
        <rect
          class="selection"
          cursor="move"
          fill="#777"
          fill-opacity="0.3"
          shape-rendering="crispEdges"
          stroke="#fff"
          style="display: none;"
        />
        <rect
          class="handle handle--e"
          cursor="ew-resize"
          style="display: none;"
        />
        <rect
          class="handle handle--w"
          cursor="ew-resize"
          style="display: none;"
        />
        <rect
          class="handle--custom"
          cursor="ew-resize"
          fill="#47DB99"
          height="15"
          rx="100"
          ry="100"
          style="opacity: 0;"
          width="15"
        />
        <rect
          class="handle--custom"
          cursor="ew-resize"
          fill="#47DB99"
          height="15"
          rx="100"
          ry="100"
          style="opacity: 0;"
          width="15"
        />
        <line
          class="bottomline"
          pointer-events="none"
          stroke="#47DB99"
          stroke-width="4"
          style="opacity: 0;"
          y1="140"
          y2="140"
        />
      </g>
    </svg>
  </div>
  <button
    class="as-btn as-btn--primary as-btn--s as-category-widget__clear"
  >
    Clear selection
  </button>
</as-histogram-widget>
`;

exports[`as-histogram-widget Rendering should render colors properly 1`] = `
<as-histogram-widget
  class="hydrated"
>
  
  <h2
    class="as-histogram-widget__header"
  >
    
  </h2>
  <p
    class="as-histogram-widget__description as-body"
  >
    
  </p>
  <div
    class="as-histogram-widget__wrapper"
  >
    <span
      class="as-histogram-widget__tooltip"
      role="tooltip"
    >
      
    </span>
    <svg
      viewBox="0 0 248 160"
    >
      <g
        class="yAxis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="end"
        transform="translate(35, 15)"
      >
        <g
          class="tick"
          opacity="1"
          transform="translate(0,125.5)"
        >
          <line
            stroke="currentColor"
            x2="205"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-10"
          >
            0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,94.25)"
        >
          <line
            stroke="currentColor"
            x2="205"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-10"
          >
            5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,63)"
        >
          <line
            stroke="currentColor"
            x2="205"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-10"
          >
            10
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,31.75)"
        >
          <line
            stroke="currentColor"
            x2="205"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-10"
          >
            15
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,0.5)"
        >
          <line
            stroke="currentColor"
            x2="205"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-10"
          >
            20
          </text>
        </g>
      </g>
      <g
        class="xAxis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(35, 140)"
      >
        <g
          class="tick"
          opacity="1"
          transform="translate(0.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="10"
          >
            0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(51.75,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="10"
          >
            10
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(103,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="10"
          >
            20
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(154.25,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="10"
          >
            30
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(205.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="10"
          >
            40
          </text>
        </g>
      </g>
      <g
        transform="translate(35, 15)"
      >
        <rect
          class="bar"
          height="0"
          style="fill: #EEFFFF;"
          width="50.25"
          x="0"
          y="125"
        />
        <rect
          class="bar"
          height="0"
          style="fill: #EEFFFF;"
          width="50.25"
          x="51.25"
          y="125"
        />
        <rect
          class="bar"
          height="0"
          style="fill: #FFAAAA;"
          width="50.25"
          x="102.5"
          y="125"
        />
        <rect
          class="bar"
          height="0"
          style="fill: #FFAAAA;"
          width="50.25"
          x="153.75"
          y="125"
        />
      </g>
      <g
        class="brush"
        fill="none"
        pointer-events="all"
      >
        <rect
          class="overlay"
          cursor="crosshair"
          height="125"
          pointer-events="all"
          width="205"
          x="35"
          y="15"
        />
        <rect
          class="selection"
          cursor="move"
          fill="#777"
          fill-opacity="0.3"
          height="125"
          shape-rendering="crispEdges"
          stroke="#fff"
          style=""
          width="102.5"
          x="35"
          y="15"
        />
        <rect
          class="handle handle--e"
          cursor="ew-resize"
          height="130"
          style=""
          width="5"
          x="135"
          y="12.5"
        />
        <rect
          class="handle handle--w"
          cursor="ew-resize"
          height="130"
          style=""
          width="5"
          x="32.5"
          y="12.5"
        />
        <rect
          class="handle--custom"
          cursor="ew-resize"
          fill="#47DB99"
          height="15"
          rx="100"
          ry="100"
          style="opacity: 1;"
          transform="translate(27.5,132.5)"
          width="15"
        />
        <rect
          class="handle--custom"
          cursor="ew-resize"
          fill="#47DB99"
          height="15"
          rx="100"
          ry="100"
          style="opacity: 1;"
          transform="translate(130,132.5)"
          width="15"
        />
        <line
          class="bottomline"
          pointer-events="none"
          stroke="#47DB99"
          stroke-width="4"
          style="opacity: 1;"
          x1="35"
          x2="137.5"
          y1="140"
          y2="140"
        />
      </g>
    </svg>
  </div>
  
</as-histogram-widget>
`;

exports[`as-histogram-widget Rendering should render properly 1`] = `
<as-histogram-widget
  class="hydrated"
>
  
  <h2
    class="as-histogram-widget__header"
  >
    Histogram Widget Example
  </h2>
  <p
    class="as-histogram-widget__description as-body"
  >
    Description for Histogram Widget
  </p>
  <div
    class="as-histogram-widget__wrapper"
  >
    <span
      class="as-histogram-widget__tooltip"
      role="tooltip"
    >
      
    </span>
    <svg
      viewBox="0 0 248 160"
    >
      <g
        class="yAxis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="end"
        transform="translate(35, 15)"
      >
        <g
          class="tick"
          opacity="1"
          transform="translate(0,125.5)"
        >
          <line
            stroke="currentColor"
            x2="205"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-10"
          >
            0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,94.25)"
        >
          <line
            stroke="currentColor"
            x2="205"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-10"
          >
            5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,63)"
        >
          <line
            stroke="currentColor"
            x2="205"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-10"
          >
            10
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,31.75)"
        >
          <line
            stroke="currentColor"
            x2="205"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-10"
          >
            15
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,0.5)"
        >
          <line
            stroke="currentColor"
            x2="205"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-10"
          >
            20
          </text>
        </g>
      </g>
      <g
        class="xAxis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(35, 140)"
      >
        <g
          class="tick"
          opacity="1"
          transform="translate(0.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="10"
          >
            0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(51.75,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="10"
          >
            10
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(103,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="10"
          >
            20
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(154.25,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="10"
          >
            30
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(205.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="10"
          >
            40
          </text>
        </g>
      </g>
      <g
        transform="translate(35, 15)"
      >
        <rect
          class="bar"
          height="0"
          style="fill: #1785FB;"
          width="50.25"
          x="0"
          y="125"
        />
        <rect
          class="bar"
          height="0"
          style="fill: #1785FB;"
          width="50.25"
          x="51.25"
          y="125"
        />
        <rect
          class="bar"
          height="0"
          style="fill: #1785FB;"
          width="50.25"
          x="102.5"
          y="125"
        />
        <rect
          class="bar"
          height="0"
          style="fill: #1785FB;"
          width="50.25"
          x="153.75"
          y="125"
        />
      </g>
      <g
        class="brush"
        fill="none"
        pointer-events="all"
      >
        <rect
          class="overlay"
          cursor="crosshair"
          height="125"
          pointer-events="all"
          width="205"
          x="35"
          y="15"
        />
        <rect
          class="selection"
          cursor="move"
          fill="#777"
          fill-opacity="0.3"
          shape-rendering="crispEdges"
          stroke="#fff"
          style="display: none;"
        />
        <rect
          class="handle handle--e"
          cursor="ew-resize"
          style="display: none;"
        />
        <rect
          class="handle handle--w"
          cursor="ew-resize"
          style="display: none;"
        />
        <rect
          class="handle--custom"
          cursor="ew-resize"
          fill="#47DB99"
          height="15"
          rx="100"
          ry="100"
          style="opacity: 0;"
          width="15"
        />
        <rect
          class="handle--custom"
          cursor="ew-resize"
          fill="#47DB99"
          height="15"
          rx="100"
          ry="100"
          style="opacity: 0;"
          width="15"
        />
        <line
          class="bottomline"
          pointer-events="none"
          stroke="#47DB99"
          stroke-width="4"
          style="opacity: 0;"
          y1="140"
          y2="140"
        />
      </g>
    </svg>
  </div>
  <button
    class="as-btn as-btn--primary as-btn--s as-category-widget__clear"
  >
    Clear selection
  </button>
</as-histogram-widget>
`;

exports[`as-histogram-widget Rendering should render selection properly 1`] = `
<as-histogram-widget
  class="hydrated"
>
  
  <h2
    class="as-histogram-widget__header"
  >
    
  </h2>
  <p
    class="as-histogram-widget__description as-body"
  >
    
  </p>
  <div
    class="as-histogram-widget__wrapper"
  >
    <span
      class="as-histogram-widget__tooltip"
      role="tooltip"
    >
      
    </span>
    <svg
      viewBox="0 0 248 160"
    >
      <g
        class="yAxis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="end"
        transform="translate(35, 15)"
      >
        <g
          class="tick"
          opacity="1"
          transform="translate(0,125.5)"
        >
          <line
            stroke="currentColor"
            x2="205"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-10"
          >
            0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,94.25)"
        >
          <line
            stroke="currentColor"
            x2="205"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-10"
          >
            5
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,63)"
        >
          <line
            stroke="currentColor"
            x2="205"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-10"
          >
            10
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,31.75)"
        >
          <line
            stroke="currentColor"
            x2="205"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-10"
          >
            15
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(0,0.5)"
        >
          <line
            stroke="currentColor"
            x2="205"
          />
          <text
            dy="0.32em"
            fill="currentColor"
            x="-10"
          >
            20
          </text>
        </g>
      </g>
      <g
        class="xAxis"
        fill="none"
        font-family="sans-serif"
        font-size="10"
        text-anchor="middle"
        transform="translate(35, 140)"
      >
        <g
          class="tick"
          opacity="1"
          transform="translate(0.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="10"
          >
            0
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(51.75,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="10"
          >
            10
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(103,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="10"
          >
            20
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(154.25,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="10"
          >
            30
          </text>
        </g>
        <g
          class="tick"
          opacity="1"
          transform="translate(205.5,0)"
        >
          <text
            dy="0.71em"
            fill="currentColor"
            y="10"
          >
            40
          </text>
        </g>
      </g>
      <g
        transform="translate(35, 15)"
      >
        <rect
          class="bar"
          height="0"
          style="fill: #47DB99;"
          width="50.25"
          x="0"
          y="125"
        />
        <rect
          class="bar"
          height="0"
          style="fill: #47DB99;"
          width="50.25"
          x="51.25"
          y="125"
        />
        <rect
          class="bar"
          height="0"
          style="fill: #1785FB;"
          width="50.25"
          x="102.5"
          y="125"
        />
        <rect
          class="bar"
          height="0"
          style="fill: #1785FB;"
          width="50.25"
          x="153.75"
          y="125"
        />
      </g>
      <g
        class="brush"
        fill="none"
        pointer-events="all"
      >
        <rect
          class="overlay"
          cursor="crosshair"
          height="125"
          pointer-events="all"
          width="205"
          x="35"
          y="15"
        />
        <rect
          class="selection"
          cursor="move"
          fill="#777"
          fill-opacity="0.3"
          height="125"
          shape-rendering="crispEdges"
          stroke="#fff"
          style=""
          width="102.5"
          x="35"
          y="15"
        />
        <rect
          class="handle handle--e"
          cursor="ew-resize"
          height="130"
          style=""
          width="5"
          x="135"
          y="12.5"
        />
        <rect
          class="handle handle--w"
          cursor="ew-resize"
          height="130"
          style=""
          width="5"
          x="32.5"
          y="12.5"
        />
        <rect
          class="handle--custom"
          cursor="ew-resize"
          fill="#47DB99"
          height="15"
          rx="100"
          ry="100"
          style="opacity: 1;"
          transform="translate(27.5,132.5)"
          width="15"
        />
        <rect
          class="handle--custom"
          cursor="ew-resize"
          fill="#47DB99"
          height="15"
          rx="100"
          ry="100"
          style="opacity: 1;"
          transform="translate(130,132.5)"
          width="15"
        />
        <line
          class="bottomline"
          pointer-events="none"
          stroke="#47DB99"
          stroke-width="4"
          style="opacity: 1;"
          x1="35"
          x2="137.5"
          y1="140"
          y2="140"
        />
      </g>
    </svg>
  </div>
  
</as-histogram-widget>
`;
